webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/domain/reports/StatusReports.scala

```diff
@@ -629,3 +627,331 @@ object NodeStatusReportSerialization {
+  import com.normation.rudder.services.policies.PolicyId
+  import com.normation.utils.DateFormaterService.json.*
+  import io.scalaland.chimney.*
+  import io.scalaland.chimney.syntax.*
+  import zio.json.*
+
+  implicit lazy val codecNodeConfigId:      JsonCodec[NodeConfigId]      = JsonCodec.string.transform(NodeConfigId.apply, _.value)
+  implicit lazy val codecReportType:        JsonCodec[ReportType]        = DeriveJsonCodec.gen
+  implicit lazy val encoderReportingLogic:  JsonEncoder[ReportingLogic]  = JsonEncoder.string.contramap(_.value)
+  implicit lazy val decoderReportingLogic:  JsonDecoder[ReportingLogic]  =
+    JsonDecoder.string.mapOrFail(s => ReportingLogic.parse(s).left.map(_.fullMsg))
+  implicit lazy val codecRunComplianceInfo: JsonCodec[RunComplianceInfo] = DeriveJsonCodec.gen
+  implicit lazy val codecPolicyModeError:   JsonCodec[PolicyModeError]   = DeriveJsonCodec.gen
+  implicit lazy val codecPolicyId:          JsonCodec[PolicyId]          = new JsonCodec(
+    JsonEncoder.chunk[String].contramap(x => Chunk(x.ruleId.serialize, x.directiveId.serialize, x.techniqueVersion.serialize)),
+    JsonDecoder.chunk[String].mapOrFail {
+      case Chunk(a, b, c) =>
+        for {
+          r <- RuleId.parse(a)
+          d <- DirectiveId.parse(b)
+          t <- TechniqueVersion.parse(c)
+        } yield PolicyId(r, d, t)
+      case x              => Left(s"Can not deserialize '${x}' into a policy ID.")
+    }
+  )
+  implicit lazy val codecOverriddenPolicy:  JsonCodec[OverriddenPolicy]  = DeriveJsonCodec.gen
+  implicit lazy val codecTechniqueVersion:  JsonCodec[TechniqueVersion]  = new JsonCodec(
+    JsonEncoder.string.contramap(_.serialize),
+    JsonDecoder.string.mapOrFail(TechniqueVersion.parse)
+  )
+
+  final case class JRunAnalysis(
+      @jsonField("k")
+      kind:                RunAnalysisKind,
+      @jsonField("eid")
+      expectedConfigId:    Option[NodeConfigId],
+      @jsonField("ecs")
+      expectedConfigStart: Option[DateTime],
+      @jsonField("exp")
+      expirationDateTime:  Option[DateTime],
+      @jsonField("expSince")
+      expiredSince:        Option[DateTime],
+      @jsonField("rt")
+      lastRunDateTime:     Option[DateTime],
+      @jsonField("rid")
+      lastRunConfigId:     Option[NodeConfigId],
+      @jsonField("rexp")
+      lastRunExpiration:   Option[DateTime]
+  ) {
+    def to: RunAnalysis = this.transformInto[RunAnalysis]
+  }
+
+  object JRunAnalysis {
+    implicit lazy val i: Iso[JRunAnalysis, RunAnalysis] = Iso.derive
+
+    // we don't want that one to be used elsewhere
+    implicit lazy val encoderRunAnalysisKind: JsonEncoder[RunAnalysisKind] = JsonEncoder.string.contramap(_.entryName)
+    implicit lazy val decoderRunAnalysisKind: JsonDecoder[RunAnalysisKind] =
+      JsonDecoder.string.mapOrFail(x => RunAnalysisKind.withNameEither(x).left.map(_.getMessage()))
+
+    implicit lazy val codecRunAnalysis: JsonCodec[JRunAnalysis] = DeriveJsonCodec.gen
```

Comment:
well, the comment is actually more general than `RunAnalysisKind`, it may just have been one of the first I wrote. All the encoders are specialized for the PostgreSQL serialisation, ie for that specific external API which is quite peculiar (short names, etc). 
Also, that was the only place where I had that type. But all in all, I think that comment just needs to be removed.